File: app/src/main/java/com/uici/repertoire/ContactAdapter.java
package com.uici.repertoire;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.uici.repertoire.databinding.ItemContactBinding;

import java.util.List;

public class ContactAdapter extends RecyclerView.Adapter<ContactAdapter.ContactViewHolder> {

    private List<Contact> contacts;
    private OnContactActionListener listener;

    public interface OnContactActionListener {
        void onCallClick(Contact contact);
        void onSmsClick(Contact contact);
        void onLocationClick(Contact contact);
        void onDeleteClick(Contact contact);
    }

    public ContactAdapter(List<Contact> contacts, OnContactActionListener listener) {
        this.contacts = contacts;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ContactViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemContactBinding binding = ItemContactBinding.inflate(
                LayoutInflater.from(parent.getContext()),
                parent,
                false
        );
        return new ContactViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ContactViewHolder holder, int position) {
        holder.bind(contacts.get(position));
    }

    @Override
    public int getItemCount() {
        return contacts.size();
    }

    class ContactViewHolder extends RecyclerView.ViewHolder {
        private ItemContactBinding binding;

        public ContactViewHolder(ItemContactBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(Contact contact) {
            binding.textViewName.setText(contact.getNom() + " " + contact.getPrenoms());
            binding.textViewPhone.setText(contact.getTelephone());

            String email = contact.getEmail().isEmpty() ? "Pas d'email" : contact.getEmail();
            binding.textViewEmail.setText(email);

            String location = contact.getLieuHabitation().isEmpty() ?
                    "Pas de localisation" : contact.getLieuHabitation();
            binding.textViewLocation.setText(location);

            binding.imageButtonCall.setOnClickListener(v -> listener.onCallClick(contact));
            binding.imageButtonSms.setOnClickListener(v -> listener.onSmsClick(contact));
            binding.imageButtonLocation.setOnClickListener(v -> listener.onLocationClick(contact));
            binding.imageButtonDelete.setOnClickListener(v -> listener.onDeleteClick(contact));
        }
    }
}

File: app/src/main/java/com/uici/repertoire/ContactAdapter.kt
package com.uici.repertoire

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.uici.repertoire.databinding.ItemContactBinding

class ContactAdapter(
    private val contacts: List<Contact>,
    private val onCallClick: (Contact) -> Unit,
    private val onSmsClick: (Contact) -> Unit,
    private val onLocationClick: (Contact) -> Unit,
    private val onDeleteClick: (Contact) -> Unit
) : RecyclerView.Adapter<ContactAdapter.ContactViewHolder>() {

    inner class ContactViewHolder(private val binding: ItemContactBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(contact: Contact) {
            binding.textViewName.text = "${contact.nom} ${contact.prenoms}"
            binding.textViewPhone.text = contact.telephone
            binding.textViewEmail.text = if (contact.email.isNotEmpty()) contact.email else "Pas d'email"
            binding.textViewLocation.text = if (contact.lieuHabitation.isNotEmpty())
                contact.lieuHabitation else "Pas de localisation"

            binding.imageButtonCall.setOnClickListener { onCallClick(contact) }
            binding.imageButtonSms.setOnClickListener { onSmsClick(contact) }
            binding.imageButtonLocation.setOnClickListener { onLocationClick(contact) }
            binding.imageButtonDelete.setOnClickListener { onDeleteClick(contact) }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactViewHolder {
        val binding = ItemContactBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return ContactViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ContactViewHolder, position: Int) {
        holder.bind(contacts[position])
    }

    override fun getItemCount(): Int = contacts.size
}
